### PR TITLE
[10.0][Stock] Added migration scripts for stock module

### PR DESCRIPTION
**Pull request by pepetreshere**
Description of the issue/feature this PR addresses: Added migration scripts to v10 for stock module

Current behavior before PR: Module stock fail migration

Desired behavior after PR is merged: Module stock migrated successfully

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by lfreeke** [comment]
@jbeficent can you have a look at this?

File: addons/stock/migrations/10.0.1.1/openupgrade_analysis_work.txt
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/openupgrade_analysis_work.txt]: Remove the TODO and put `Done: explanation of what you have done`

File: addons/stock/migrations/10.0.1.1/pre-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/pre-migration.py]: Use `copyright` full word

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Use `copyright` full word

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Put this at the end of the file by general convention

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: According PEP257, method documentation goes with triple double quota, not simple, and first line start in the same one as the initial quotas.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Don't let blank lines inside a method. They're useless and are less readable.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Don't overpass 80 chars limit

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Pre-load this references. Don't load them on each loop.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: You can use ORM API here instead of dealing with the ugly SQL.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Don't be too verbose. This line can be removed.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Use one quote when the query is very simple.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: You have forgotten to put Done on the work analysis for this.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: Again, using ORM can be simple and the same in speed (more or less).

File: addons/stock/migrations/10.0.1.1/pre-migration.py
Comment (pedrobaeza) [inline addons/stock/migrations/10.0.1.1/pre-migration.py]: Use variables to store the list of column renames, copies, and so on, the same as in https://github.com/OCA/OpenUpgrade/pull/839/files#diff-3cdc4348ce357d41a6908e42cadbca5aR8

Group also all the copies in one.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Good work being your first submission to OpenUpgrade. Please follow my remarks.

**Comment by pepetreshere** [comment]
Thanks for your remarks @pedrobaeza!, we will work on this by next monday

File: addons/stock/migrations/10.0.1.1/openupgrade_analysis_work.txt
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/openupgrade_analysis_work.txt]: is post-migration

File: addons/stock/migrations/10.0.1.1/openupgrade_analysis_work.txt
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/openupgrade_analysis_work.txt]: the replacement should be done in post-migration

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: this should be removed. the 'manufacture' case should be into the migration of mrp module

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: ''' → """

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: → <code>env.cr.fetchone()</code>

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: comments should be in a separated line, and also there should be a space between <code>#</code> and the words

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: → procurement

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: the query should be compacted in one line

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: field <code>package_id</code> can be set, with a witty query, if you relate the move with the stock_quant

File: addons/stock/migrations/10.0.1.1/pre-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/pre-migration.py]: this replacement should be done in the post-migration

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: between the defs should be two empty lines

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: pass the <code>cr</code> instead of the <code>env</code> if the def only use the <code>cr</code>

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]
I just sent you a PR (https://github.com/TRESCLOUD/OpenUpgrade/pull/2) that includes all adjustments I requested here

**Comment by pepetreshere** [comment]
Thanks @pedrobaeza and @mreficent for your remarks, I am still doing some tests today and I will likely push a few more commits.

**Comment by pepetreshere** [comment]
@mreficent I have included some minor changes for the migration of stock module, and I think we can merge it now. 

I have included again the case for mrp.picking_type_manufacturing with a None option if it is missing (line 24 of post-migration.py), I believe it might be much easier to migrate that field in stock module rather than in mrp module (this change is not even detected in the analysis file of mrp). If you insist I can remove it again, what do you think?. Thanks in advance for your advise.

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: put a space between the comma and the False, and also remember to not overpass 80 chars limit by line in all the code

File: addons/stock/migrations/10.0.1.1/post-migration.py
Comment (MiquelRForgeFlow) [inline addons/stock/migrations/10.0.1.1/post-migration.py]: flake8 wants the whitespaces at the end of the line to be removed. there are 9 lines affected.

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

**Comment by pepetreshere** [comment]
@pedrobaeza  thanks for your fixes, this error appears again in travis:
`2017-05-31 14:43:01,815 40924 ERROR openupgrade OpenUpgrade: External ID not found in the system: mrp.picking_type_manufacturing`

I think that to avoid this missing ID we should keep the the False option in the method [env.ref method](https://github.com/OCA/OpenUpgrade/pull/837/commits/cc65573525df09124d8afae699eefcf58d6fde87), or alternatively move this part to the migration script of mrp (which is not advisable from my point of view as detailed above).

So I have included a solution in [this PR](https://github.com/TRESCLOUD/OpenUpgrade/pull/3), please comment if you agree with the resolution as I don't want to overwrite one line of your last commit without your agreement.

**Comment by pedrobaeza** [comment]
Oh, I was working on that and other missing things right now. Well, I remove that part and continue.

**Comment by pepetreshere** [comment]
Hello @pedrobaeza, just to make sure I understood correctly. I understood that you will make the fix altogether with other fixes, so I have not merged my fix into this branch. If you like me to merge [my fix ](https://github.com/TRESCLOUD/OpenUpgrade/pull/3/files) please let me know.

**Comment by pedrobaeza** [comment]
Yeah, I'm making them, but today and tomorrow there is the Odoo days in Spain (https://odoospain.odoo.com/event/jornadas-odoo-2017-2017-06-01-2017-06-02-5), so I'm quite busy. I will try to work tomorrow or on Monday.

**Comment by pedrobaeza** [comment]
@StefanRijnhart the same behavior is happening again with Travis (infinite loop), so the reason should be another one. I'll check it later.

**Comment by StefanRijnhart** [comment]
@pedrobaeza I see, thanks for the heads up. Why is the generic coa set to uninstallable?

**Comment by pedrobaeza** [comment]
That was another try before throwing this error because it tries to create some invoices without account_id. I see that this is not the ideal solution anyway. I'll make other tries later.

**Comment by hbrunn** [comment]
@pepetreshere please rebase your branch on current 10.0, then we should get a better picture from travis

**Comment by pedrobaeza** [comment]
@hbrunn I'm the one that is handling with this PR for having a green branch, but you are decided to put me in troubles :stuck_out_tongue:

**Comment by hbrunn** [comment]
sorry @pedrobaeza, I wasn't aware I'm interfering with your work here. Will cease operations for today then

**Comment by pedrobaeza** [comment]
It was a joke, Holger. Most of the PRs shouldn't be a problem, but it's better to wait to have a green branch before continuing merging more possible problems (for example, we merged sale_stock before having stock module and thus we can't get a green branch until get this one correct).

**Comment by hbrunn** [comment]
looks like stock's migration succeeds now and https://travis-ci.org/OCA/OpenUpgrade/builds/240091842#L4133 is the problem. Should we add a [force deps](https://github.com/OCA/OpenUpgrade/blob/10.0/odoo/modules/graph.py#L88) config entry to make `base_action_rule` depend on all the modules the demo data refers to? Then we should be rid of this problem too.

**Comment by pepetreshere** [comment]
Hello @hbrunn, done with the rebase of the branch, let me know if anything else is needed.

**Comment by pepetreshere** [comment]
Hello @pedrobaeza and @hbrunn , I have tried to rebase but I find everything is up to date as shown in below image. In order to not block your progress I have added you guys as collaborators with write access to this branch (hope you don't mind) as I need your support to make this rebase.

capture 2017-06-02 at 18 01 58

**Comment by hbrunn** [comment]
@pepetreshere thanks for the invitation, but @pedrobaeza already did it, that's why you don't see a difference. Now if we just fix https://github.com/OCA/OpenUpgrade/pull/837#issuecomment-306588914 too, this will be green, the whole branch will be green and we'll be super happy

**Comment by pedrobaeza** [comment]
I'm on it, so I expect to get it through the day.

**Comment by pedrobaeza** [comment]
FYI, all the PR closing/opening activity is because Travis don't set correctly $TRAVIS_COMMIT variable on rebase or forced push operations, so the test is not correct. This is the way I have found for triggering a correct test.

**Review by MiquelRForgeFlow — APPROVED** [review]
It's green.

**Comment by pedrobaeza** [comment]
And finally we have it!! :tada: :tada: :tada:

@hbrunn what I have done instead of force-deps, which requires to search for all the declared action rules, is to simply not load non existing models. Done in 9314ff87188da31d1263b6c3bf7bb4f3d2485ac0.

@StefanRijnhart I had to set as not installable l10n_generic_coa and l10n_us, or an error comes which says that the account_id of an invoice can't be null. Maybe it's because account is not yet migrated? We can activate them on the PR to see, but for now this is the only solution I see for having a green branch.

Please give your final blessings them and we merge this for fixing the branch.

**Review by pedrobaeza — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
@pedrobaeza thanks for getting this green. Should we revert this hack and keep the red branch while working on the account migration?

**Comment by StefanRijnhart** [comment]
This could then be merged, of course.

**Comment by hbrunn** [comment]
I'd suggest to merge this without the fix and hurry up with migrating hr_contract: https://github.com/odoo/odoo/search?utf8=%E2%9C%93&q=model%3D%22base.action.rule%22+language%3Axml&type=Code

**Comment by StefanRijnhart** [comment]
@hbrunn you write "I'd suggest to merge this without the fix" 

but I don't know what you mean exactly. To be precise myself, I think the change to base_action_rule is fine, but I would rather not set modules uninstallable.

**Comment by hbrunn** [comment]
@StefanRijnhart I feel uncomfortable patching upstream code where it's not absolutely necessary, so I'd prefer not to have 9314ff87188da31d1263b6c3bf7bb4f3d2485ac0 in OpenUpgrade. You know from painful experience in #895 that every change we apply is a possible source of conflicts.

My proposal is: Revert 9314ff87188da31d1263b6c3bf7bb4f3d2485ac0, merge https://github.com/OCA/OpenUpgrade/pull/897, then this should be green too, and we're happy

**Comment by StefanRijnhart** [comment]
@hbrunn I can agree with that, yes.

**Comment by StefanRijnhart** [comment]
Wait, you want to keep setting modules to uninstallable?

**Comment by pedrobaeza** [comment]
@hbrunn I don't think the change in base_action_rule will conflict in the future, but anyway, have you checked if there's any other non migrated module that declares action rules?

**Comment by hbrunn** [comment]
@StefanRijnhart setting modules to uninstallable: Now that I looked at the code for the first time, no, I don't want that either
@pedrobaeza I think https://github.com/odoo/odoo/search?utf8=%E2%9C%93&q=model%3D%22base.action.rule%22+language%3Axml&type=Code makes reasonably sure that's the last module causing trouble

**Comment by pedrobaeza** [comment]
We still have problems with website module that don't anticipate before. Should I fix it here or do you prefer to merge this and create another PR fixing website?

**Comment by StefanRijnhart** [comment]
We can merge, if you remove the change of setting the coa modules uninstallable.

**Comment by pedrobaeza** [comment]
Removing that commit, I won't still be able to get a green branch although I fix website part. Can't we let the commit and revert later when dealing with account migration?

**Comment by pedrobaeza** [comment]
OK, finally I'm going to translate my greenifying efforts to the account migration PR, so I have removed the commit that makes uninstallable the l10n_* modules and merging this one although in red. Then, I'll wait for website PR and rebase everything to work on the final solution at account level.

**Comment by JordiBForgeFlow** [comment]
:+1:

**Comment by pepetreshere** [comment]
👍 :)

**Pull request merged**
